Content.Server/Ghost/Roles/GhostRoleSystem.cs

```diff
@@ -346,6 +352,44 @@ private void OnSpawnerTakeRole(EntityUid uid, GhostRoleMobSpawnerComponent compo
             args.TookRole = true;
         }
 
+        // Delta-V code for custom spawner type, that'll spawn a players currently selected character instead of an entity
+        private void OnSpawnerTakeCharacter(EntityUid uid, GhostRoleCharacterSpawnerComponent component, ref TakeGhostRoleEvent args)
+        {
+            if (!TryComp(uid, out GhostRoleComponent? ghostRole) ||
+                ghostRole.Taken)
+            {
+                args.TookRole = false;
+                return;
+            }
+
+            var character = (HumanoidCharacterProfile) _prefs.GetPreferences(args.Player.UserId).SelectedCharacter;
+
+            var mob = _entityManager.System<StationSpawningSystem>()
+                .SpawnPlayerMob(coordinates: Transform(uid).Coordinates, profile: character, entity: null, job: null, station: null);
```

Comment:
does something like this not work? it's weird defining each variable to assign to
```suggestion
                .SpawnPlayerMob(Transform(uid).Coordinates, character, null, null, null);
```